File: src/main/java/org/fletchly/genius/data/model/Message.java
package org.fletchly.genius.data.model;

import lombok.Builder;
import lombok.Getter;

/**
 * @param role "user" | "assistant" | "tool"
 */
@Builder
public record Message(int id, int conversationId, String role, String content, long timestamp) {
    public enum Role {
        USER("user"),
        TOOL("tool"),
        ASSISTANT("assistant"),
        SYSTEM("system");

        @Getter
        private final String role;

        Role(String role) {
            this.role = role;
        }
    }
}

File: src/main/kotlin/io/fletchly/genius/conversation/model/Message.kt
package io.fletchly.genius.conversation.model

import kotlinx.serialization.Serializable

@Serializable
data class Message(
    val content: String,
    val role: String
) {
    companion object Role {
        const val SYSTEM = "system"
        const val USER = "user"
        const val ASSISTANT = "assistant"
        const val TOOL = "tool"
    }
}